avocado/core/job.py

```diff
@@ -564,6 +570,9 @@ def run(self):
             self.log.error('Report bugs visiting %s', _NEW_ISSUE_LINK)
             return exit_codes.AVOCADO_FAIL
         finally:
+            for ext in self.job_pre_post_dispatcher.extensions:
+                if hasattr(ext.obj, 'post'):
+                    ext.obj.post(self)
```

Comment:
My point was that the Pre and Post jobs are such a simple interface, that I don't see the additional benefit in making it so complex. I understand the benefit in complex systems, but the current plugin system is compound of more-less simple hooks.

Anyway I see you're really in favor of this and the only drawback I have here is that we'd have to remember always to ask users to do both:

1. Use the right abs class(es)
2. Implement the correct methods

The later is consequence of the first, but if they do it in the other way around, it'd still work and we'd have to ask them to add the right inheritance.